Task 2/2.1/2.1.2/Polygons.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace _2._1._2
+{
+    public abstract class Polygon : Figure //многоугольник
+    {
+        public Point[] points { private set; get; }
+        public Polygon(string name, params Point[] points) : base(name)
+        {
+            this.points = points;
+            if (points.Length > points.Distinct().Count())
+            {
+                throw new Exception("Polygon can't contain the same points");
+            }
+        }
+
+        public override double GetLength()
+        {
+            double length = 0;
+            if (points.Length == 2) return points[0].GetDistance(points[1]);
+            for (int i = 0; i < points.Length - 1; i++)
+                {
+                    length += points[i].GetDistance(points[i+1]);
+                }
+            length += points[points.Length - 1].GetDistance(points[0]);
+            return length;
+        }
+
+    }
+
+    public class Line : Polygon //линия
+    {
+        public Line(Point one, Point two) : base("Line", one, two) { }
+
+        public override double GetArea()
+        {
+            return 0;
+        }
+
+        public override string ToString()
+        {
+            return name + " - Point one: " + points[0] + ", two: " + points[1];
+        }
+
+        public bool isPerpendicular(Line argument)
```

Comment:
PascalCase for types and public members, camelCase for private fields and local variables